e2e/tables.spec.ts

```diff
@@ -0,0 +1,430 @@
+import { expect, test } from '@playwright/test'
+import { testAuth } from './fixtures/authenticatedUserPages'
+import { deleteEntity, generateEntityName } from './helpers/entities'
+import {
+  setupProject,
+  teardownProjectsAndFileHandles,
+} from './helpers/setupTeardown'
+import {
+  ColumnSchemaConfig,
+  addColumnsFromColumnSchemaConfig,
+  expectTableSchemaCorrect,
+  openTableColumnSchema,
+} from './helpers/tableColumnSchemaEditor'
+import {
+  TableDataConfig,
+  confirmTableDataDimensions,
+  enterTableData,
+  enterTableValue,
+  expectTableDataCorrect,
+  expectTableItemNumberCorrect,
+  expectTablePageLoaded,
+  expectTablesPageLoaded,
+  getDefaultQuery,
+  getTableEditorRows,
+  getTablesPath,
+  openTableEditor,
+  runQuery,
+  saveTableDataChanges,
+  toggleQueryEditor,
+} from './helpers/tables'
+import {
+  dismissAlert,
+  getAdminPAT,
+  goToDashboardPage,
+  toggleIntoExperimentalMode,
+} from './helpers/testUser'
+import { Project } from './helpers/types'
+
+let userProject: Project
+let fileHandleIds: string[] = []
+let tableEntityId: string | undefined = undefined
+
+const noTablesText =
+  'There are no Tables associated with this project. Any Tables you create in this project will appear here'
+
+test.describe('Tables', () => {
+  testAuth.beforeAll(async ({ browser, storageStatePaths }) => {
+    userProject = await setupProject(browser, 'swc-e2e-user', storageStatePaths)
+  })
+
+  testAuth.afterAll(async ({ browser }) => {
+    test.slow()
+    if (tableEntityId) {
+      const page = await browser.newPage()
+      await deleteEntity(tableEntityId, true, getAdminPAT(), page)
+      await page.close()
+    }
+    if (userProject.id) {
+      await teardownProjectsAndFileHandles(
+        browser,
+        [userProject],
+        fileHandleIds,
+      )
+    }
+  })
+
+  testAuth('should allow table CRUD', async ({ userPage }) => {
+    test.slow()
+
+    const tableName = generateEntityName('table')
+    const tableDescription = 'table test description'
+
+    const initialColumnsSchemaConfig: ColumnSchemaConfig[] = [
+      { name: 'item', type: 'String', size: 20 },
+      { name: 'quantity', type: 'Integer', defaultValue: '500' },
+      // { name: 'on_sale', type: 'Boolean', defaultValue: 'false' }, - currently broken, SWC-6630
```

Comment:
Thanks for writing + commenting these cases and mentioning the associated ticket! I attached comments to each ticket as a reminder (for me) to come back through and fix all of these!

Reply:
Thank you!!